dotnet/WPF/VisualLayerIntegration/VisualLayerIntegration/BarGraphUtility/BarBrushHelper.cs

```diff
@@ -30,111 +30,90 @@
 namespace BarGraphUtility
 {
     // Create brushes to fill the bars. 
-    public class BarBrushHelper
+    sealed class BarBrushHelper
     {
-        private Compositor compositor;
-        private Random rand;
+        private readonly Compositor _compositor;
+        private readonly Random _rand = new Random();
 
         public BarBrushHelper(Compositor c)
         {
-            compositor = c;
-            rand = new Random();
+            _compositor = c;            
         }
 
-        internal CompositionBrush[] GenerateSingleColorBrush(int numBrushes, Color color)
+        internal CompositionBrush GenerateSingleColorBrush(Color color)
         {
-            var brushes = new CompositionBrush[numBrushes];
-            for (int i = 0; i < numBrushes; i++)
-            {
-                brushes[i] = compositor.CreateColorBrush(color);
-            }
-
-            return brushes;
+            return _compositor.CreateColorBrush(color);
         }
 
-
         internal CompositionBrush[] GenerateRandomColorBrushes(int numBrushes)
         {
-            var max = byte.MaxValue + 1; // 256
             var brushes = new CompositionBrush[numBrushes];
             for (int i = 0; i < numBrushes; i++)
             {
-                byte r = Convert.ToByte(rand.Next(max));
-                byte g = Convert.ToByte(rand.Next(max));
-                byte b = Convert.ToByte(rand.Next(max));
-                Color c = Color.FromArgb(Convert.ToByte(255), r, g, b);
-                brushes[i] = compositor.CreateColorBrush(c);
+                var rgb = new byte[3];
+                _rand.NextBytes(rgb);
+                var c = Color.FromArgb(255, rgb[0], rgb[1], rgb[2]);
+                brushes[i] = _compositor.CreateColorBrush(c);
             }
 
             return brushes;
         }
 
-        internal CompositionBrush[] GeneratePerBarLinearGradient(int numBrushes, List<Color> colors)
+        internal CompositionBrush GenerateLinearGradient(Color[] colors)
```

Comment:
This doesn't return a gradient, and it doesn't generate, so the name should be changed to something like CreateLinearGradientBrushes.